### PR TITLE
trivial: Exclude directories from second stage bootloader checks

### PR DESCRIPTION
**Pull request by superm1**
Fixes: #8224

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-context.c [hunk 1]
```diff
@@ -1921,7 +1921,8 @@ fu_context_get_esp_files_for_entry(FuContext *self,
 			g_string_replace(filename2, shim_name, grub_name, 1);
 		}
 		g_debug("check for 2nd stage bootloader: %s", filename2->str);
-		if (g_file_test(filename2->str, G_FILE_TEST_EXISTS)) {
+		if (!g_file_test(filename2->str, G_FILE_TEST_IS_DIR) &&
+		    g_file_test(filename2->str, G_FILE_TEST_EXISTS)) {
```
Comment (hughsie) [inline libfwupdplugin/fu-context.c:1925]: I think there's an IS_FILE that does both, no?

Comment (superm1) [reply libfwupdplugin/fu-context.c:1925]: `G_FILE_TEST_IS_REGULAR `.  Is that good you think here?

**Comment by cgzones** [comment]
Works for me, thanks :+1:

**Review by hughsie — APPROVED** [review]
Thanks both!

**Pull request merged**